src/babashka/curl.clj

```diff
@@ -156,9 +156,10 @@
         is (read-then-unread is)
         err (:err opts)
         stream? (identical? :stream (:as opts))
-        [body err] (if stream?
-                     [is err]
-                     [(slurp is) (slurp err)])
+        process (:proc opts)
+        [body err exit] (if stream?
+                          [is err (delay (.waitFor ^java.lang.Process process))]
+                          [(slurp is) (slurp err) (.waitFor ^java.lang.Process process)])
```

Comment:
@borkdude Per your suggestion, using a `delay` for the exit code in the case of `:as :stream`. I'm wondering if this would be confusing and maybe it should always be a delay?

Reply:
We already have a difference here between the types for when you request a stream, so I think it might be ok.

Reply:
Btw, there is a function called `force` in `clojure.core`:

```
$ clj
Clojure 1.10.1
user=> (force 1)
1
user=> (force (delay 1))
1
```
So people can call that on the result and be ok with both "types".

Reply:
Oh cool. I didn't know about `force`.